paddle/fluid/pir/transforms/onednn/conv_elementwise_add_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,405 @@
+      : conv_name_(conv_name), fused_conv_name_(fused_conv_name) {}
+
+  std::string name() const override {
+    return "FusedConvBiasElementwiseAddPattern";
+  }
+
+  uint32_t benefit() const override { return 2; }
+
+  void operator()(paddle::drr::DrrPatternContext *ctx) const override {
+    paddle::drr::SourcePattern pat = ctx->SourcePattern();
+    const auto &conv =
+        pat.Op(conv_name_,
+               {{
+                   {"strides", pat.Attr("strides")},
+                   {"paddings", pat.Attr("paddings")},
+                   {"padding_algorithm", pat.Attr("padding_algorithm")},
+                   {"dilations", pat.Attr("dilations")},
+                   {"groups", pat.Attr("groups")},
+                   {"data_format", pat.Attr("data_format")},
+               }});
+
+    const auto &add = pat.Op(paddle::dialect::AddOp::name());
+    conv({&pat.Tensor("input"),
+          &pat.Tensor("filter"),
+          &pat.Tensor("bias"),
+          &pat.Tensor("__@input_none_tensor@__")},
+         {&pat.Tensor("conv2d_out")});
+
+    pat.Tensor("add_out") =
+        add(pat.Tensor("conv2d_out"), pat.Tensor("residual_param"));
+    pat.RequireNativeCall(
+        [](const paddle::drr::MatchContext &match_ctx) -> bool {
+          auto padding_algorithm =
+              match_ctx.Attr<std::string>("padding_algorithm");
+          if (padding_algorithm != "EXPLICIT" && padding_algorithm != "SAME" &&
+              padding_algorithm != "VALID") {
+            return false;
+          }
+          auto groups = match_ctx.Attr<int>("groups");
+          if (groups < 1) {
+            return false;
+          }
+          auto data_format = match_ctx.Attr<std::string>("data_format");
+          if (data_format != "NCHW" && data_format != "AnyLayout") {
+            return false;
+          }
+          return true;
+        });
+    paddle::drr::ResultPattern res = pat.ResultPattern();
+
+    const auto &fused_conv2d_add =
+        res.Op(fused_conv_name_,
+               {{
+                   {"strides", pat.Attr("strides")},
+                   {"paddings", pat.Attr("paddings")},
+                   {"padding_algorithm", pat.Attr("padding_algorithm")},
+                   {"dilations", pat.Attr("dilations")},
+                   {"groups", pat.Attr("groups")},
+                   {"data_format", pat.Attr("data_format")},
+                   {"mkldnn_data_type", res.StrAttr("float32")},
+                   {"fuse_activation", res.StrAttr("")},
+                   {"fuse_residual_connection", res.BoolAttr(true)},
+                   {"force_fp32_output", res.BoolAttr(false)},
+                   {"fuse_alpha", res.Float32Attr(0.0f)},
+                   {"fuse_beta", res.Float32Attr(0.0f)},
+                   {"scale_in", res.Float32Attr(1.0f)},
+                   {"scale_out", res.Float32Attr(1.0f)},
+                   {"scale_in_eltwise", res.Float32Attr(1.0f)},
+                   {"scale_weights", res.VectorFloatAttr({1.0f})},
```

Comment:
From my perspective, if the entrance op is a fused_conv, then when you create a new fused_conv with add, the above "fused_attributes" should be grasped from the former fused_op instead of just setting default value, since you don't know if these attrs are already set or not.

Reply:
It's safe if we have only "conv_bias_fuse_pass" and "conv_eltwise_add_fuse_pass" here, but not if we have other passes at former place, especially when they will output fused_conv as well.

Reply:
yes, since I notice in the previous pass order, just conv_bias_fuse_pass before this pass to set these attribute. Therefore, I think there is no previous pass that will affect the value here. I will check and think again.